workerpool/workerpool.go

```diff
@@ -100,37 +119,35 @@ func New[T any](concurrency int64) *Pool[T] {
 //
 // Never call Do from within a Do function because it will deadlock.
 func (p *Pool[T]) Do(ctx context.Context, fn WorkFunc[T]) error {
-	// Do not enqueue new work if the worker pool is stopped.
+	i := atomic.AddInt64(&p.i, 1)
+
 	if p.isStopped() {
+		p.appendResult(i, *new(T), ErrStopped)
 		return ErrStopped
```

Comment:
Should we consider removing the error response from this func since we will get all the errors when Done is called? That will be a bit nicer on the consumer side.

Reply:
You might want to stop queuing new work on the client based on the fact that the worker has stopped.

Reply:
Ack